src/heitech.zer0mqXt.core/Patterns/Socket.cs

```diff
@@ -87,21 +87,41 @@ public Socket(SocketConfiguration configuration)
 
         #region Response / Server
         private ManualResetEvent eventHandle;
-
         private ResponseSocket responseSocket;
         private NetMQ.NetMQPoller poller;
         private EventHandler<NetMQSocketEventArgs> receiveHandler;
         private readonly object concurrencyToken = new object();
         private bool responderIsSetup = false;
         private bool respondingIsActive = false;
 
+        ///<summary>
+        /// Register async Callback on the Respond Action at the server
+        ///</summary>
+        public void RespondAsync<T, TResult>(Func<T, Task<TResult>> factory, CancellationToken cancellationToken = default)
```

Comment:
That is what I did first, but then the method call is ambigous because Type TResult :class could also be a task<TSomeType> so this is how other libraries often solve it. (There is plenty of Async methods that return void) Yet I agree it feels kinda weird.

What do you propose instead? I´d like to keep the name similar to Respond, so Intellisense will assist you when you are trying to find your way around the library etc.


- RespondWithAsyncCallback
- AsyncRespond
- AsyncResponder
- ??

Reply:
Ok so until we find a better name I would stick with the one we got for now. The thing is, for the ISocket/IBus interface where the public API resides, we must decide on a different name. for the internal type it would be nice if it is consistent, but as it is internal is not of utmost importance. 

Hence I will create an issue for this.